src/error.rs

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Error management
+
+use std::fmt::Display;
+use std::process::exit;
+
+// Define a trait with the exit_if_error method
+pub trait ExitIfError<T, E> {
+    /// Call the error code and exit the process if the result is an error.
+    /// Otherwise, return the value.
+    ///
+    /// # Arguments
+    /// * `self` - The result to check
+    /// * `err_handler` - The error handler to call if the result is an error
+    ///
+    /// # Returns
+    /// The value if the result is Ok
+    fn exit_if_error<F: FnOnce(E)>(self, err_handler: F) -> T;
+
+    /// Call the error code and exit the process with the given code if the
+    /// result is an error.
+    ///
+    /// # Arguments
+    /// * `self` - The result to check
+    /// * `code` - The exit code to use if the result is an error
+    /// * `err_handler` - The error handler to call if the result is an error
+    ///
+    /// # Returns
+    /// The value if the result is Ok
+    #[allow(dead_code)]
+    fn exit_with_code_if_error<F: FnOnce(E)>(self, code: i32, err_handler: F) -> T;
+}
+
+// Implement the trait for all Result<T, E> where E is an error.
+impl<T, E: Display> ExitIfError<T, E> for Result<T, E> {
+    /// Call the error code and exit the process if the result is an error.
+    /// Otherwise, return the value.
+    ///
+    /// # Arguments
+    /// * `self` - The result to check
+    /// * `err_handler` - The error handler to call if the result is an error
+    ///
+    /// # Returns
+    /// The value if the result is Ok
+    fn exit_if_error<F: FnOnce(E)>(self, err_handler: F) -> T {
```

Comment:
These are quite nice.  Thanks for adding them!

Reply:
It's better than what we had before, but there are still a few things to do in a future PR.